FinancialTracker/ViewControllers/HomeViewController.swift

```diff
@@ -51,95 +54,112 @@ class HomeViewController: UIViewController {
         }
     }
     
+    private func expenseOrIncome(start: Date, end: Date) -> [Transaction] {
+        var data: [Transaction] = []
+        if expenseOrIncomeSegmentedControl.selectedSegmentIndex == 0 {
+            data = start.transactionBetweenTwoDates(till: end, data: authManager?.currentUser?.expenses ?? [])
+        } else {
+            data = start.transactionBetweenTwoDates(till: end, data: authManager?.currentUser?.incomes ?? [])
+        }
+        return data
+    }
+    
+    @IBAction func expenseOrIncomeControlDidChange(_ sender: UISegmentedControl) {
+        dividerControlDidChange(expenseDividerSegmentedControl)
+    }
+    
     @IBAction func dividerControlDidChange(_ sender: UISegmentedControl) {
```

Comment:
Smart use of `expenseOrIncomeControlDidChange` and `dividerControlDidChange` but you should still use them separately. Or it'll best to create a separate function with a suitable name that handles `expenseOrIncomeControlDidChange` and `dividerControlDidChange`